File: tests/instructions/network/addToNetwork.ts
import * as anchor from "@coral-xyz/anchor";
import { Program } from "@coral-xyz/anchor";
import {Network} from "../../../target/types/network"
const provider = anchor.AnchorProvider.env();
anchor.setProvider(provider);
const program = anchor.workspace.Network as Program<Network>;

export const addToNetwork =  async (
  payer: anchor.web3.PublicKey,
  authority: anchor.web3.PublicKey,
  nodeAuthority: anchor.web3.PublicKey,
  childNetworkAccount: anchor.web3.PublicKey,
  networkAccount: anchor.web3.PublicKey,
  child: anchor.web3.PublicKey,
  joinEscrow: anchor.web3.PublicKey,
  node: anchor.web3.PublicKey,
) => {
  const [linkData] = anchor.web3.PublicKey.findProgramAddressSync(
    [Buffer.from("link_data"), node.toBuffer(), child.toBuffer(), authority.toBuffer()],
    program.programId
  )

  // Create reputation account
  const addToNetworkTx = await program.methods.addToNetwork
    (
    )
    .accounts({
      payer, 
      authority,
      nodeAuthority,
      child,
      networkAccount,
      node,
      childNetworkAccount,
      joinEscrow,
      linkData,
    })
    .rpc();
  return {addToNetworkTx,linkData} 
}

File: programs/network/src/instructions/add_to_network.rs
use anchor_lang::prelude::*;
use crate::state::{network::Network, node::Node,link_data::LinkData, join_escrow::JoinEscrow};
pub fn add_to_network(ctx: Context<AddToNetwork>,init_time: String) -> Result<()> {
    ctx.accounts.link_data.initialize_link(ctx.accounts.authority.key(), ctx.accounts.child.key(),ctx.accounts.network_account.key(), ctx.accounts.child_network_account.key(), ctx.accounts.node.key(),init_time)?;
    Ok(())
}

#[derive(Accounts)]
pub struct AddToNetwork<'info>{
    #[account(mut)]
    payer: Signer<'info>,
    authority: SystemAccount<'info>,
    node_authority: SystemAccount<'info>,
    #[account(mut)]
    child: SystemAccount<'info>,
    #[account(seeds=[b"network", authority.key.as_ref()], bump ,)]
    network_account: Account<'info, Network>, 
    #[account(seeds=[b"node", node_authority.key.as_ref()], bump ,)]
    node: Account<'info, Node>, 
    #[account(seeds=[b"network", child.key.as_ref()], bump ,)]
    child_network_account: Account<'info, Network>,
    #[account(mut, close=child, has_one=child, constraint = authority.key() == join_escrow.parent , seeds=[b"join_escrow", child.key.as_ref()], bump , )]
    join_escrow: Account<'info, JoinEscrow>,
    #[account(init, payer=payer, seeds=[b"link_data",node.key().as_ref(),child.key.as_ref(), authority.key.as_ref()], bump, space=LinkData::INIT_SPACE)]
    link_data: Account<'info,LinkData >,
    system_program: Program<'info, System>
}


